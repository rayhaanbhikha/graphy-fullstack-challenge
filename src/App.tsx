import React, { useState } from 'react';
import styled from 'styled-components';

import { Annotations } from './components/Annotations/Annotations'

const Container = styled.div`
  width: 100vw;
  height: 100vh;
`;

const Coords = styled.div`
  color: #aaa;
`;
export interface Coord {
  x: number;
  y: number;
}

const App = () => {
  const [coords, setCoords] = useState<Coord>({ x: 0, y: 0 });

  const handleMouseMove = (ev: React.MouseEvent) => {
    const { pageX: x, pageY: y } = ev;
    setCoords({ x, y });
  };

  return (
    <Container onMouseMove={handleMouseMove}>
      <Coords>{coords.x},{coords.y}</Coords>
      <Annotations coord={coords} />
    </Container>
  );
};

export default App;
