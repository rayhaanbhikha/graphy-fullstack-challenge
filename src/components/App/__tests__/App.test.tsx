import React from 'react';
import { render } from '@testing-library/react'
import App from '../App';
import 'jest-styled-components';

describe('App', () => {

  it('should have the default styles', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  })

});
