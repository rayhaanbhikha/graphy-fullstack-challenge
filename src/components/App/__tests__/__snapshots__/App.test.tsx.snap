// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should have the default styles 1`] = `
<DocumentFragment>
  .c1 {
  position: inherit;
  height: inherit;
  width: inherit;
}

.c0 {
  width: 100vw;
  height: 100vh;
}

<div
    class="c0"
  >
    <div
      class="c1"
      data-testid="annotations-wrapper"
    />
  </div>
</DocumentFragment>
`;
